docs/rackhd/install_os.rst

```diff
@@ -0,0 +1,128 @@
+=================== ================ ============ ============================================
+version             String           **required** The version number of target OS that needs to install. **NOTE**: For Ubuntu, the version should be 'trusty' not '14.04'
+repo                String           **required** The OS repository address, currently only supports HTTP
+rootPassword        String           *optional*   The password for the OS root account, default *rootPassword* is  **"RackHDRocks!"**
+hostname            String           *optional*   The hostname for target OS, default *hostname* is **"localhost"**
+domain              String           *optional*   The domain for target OS, default *domain* is **"rackhd"**
+users               Array            *optional*   If specified, this contains an array of objects, each object contains the user account information that will be created after OS installation. 0, 1, or multiple users could be specified.  If *users* is omitted, null or empty, no user will be created.
+dnsServers          Array            *optional*   If specified, this contains an array of string, each elements is the Domain Name Server, the first one will be primary, others are alternative.
+networkDevices      Array            *optional*   The static IP setting for network devices after OS installation. If it is omitted, null or empty, RackHD will not touch any network devices setting, so all network devices remain at the default state (usually default is DHCP).If there is multiple setting for same device, RackHD will choose the last one as the final setting, both ipv4 and ipv6 are supported here.
+rootSshKey          String           *optional*   The SSH key that will be appended to target OS.
+installDisk         String/Number    *optional*   *installDisk* is to specify the target disk which the OS will be installed on. It can be a string or a number. **For string**, it is a disk path that the OS can recongize, its format varies with OS. For example, "/dev/sda" or "/dev/disk/by-id/scsi-36001636121940cc01df404d80c1e761e" for CentOS/RHEL, "t10.ATA_____SATADOM2DSV_3SE__________________________20130522AA0990120088" or "naa.6001636101840bb01df404d80c2d76fe" or "mpx.vmhba1:C0:T1:L0" or "vml.0000000000766d686261313a313a30" for ESXi. **For number**, it is a RackHD generated disk identifier (it could be obtained from "driveId" catalog). If *installDisk* is omitted, RackHD will assign the default disk by order: SATADOM -> first disk in "driveId" catalog -> "sda" for Linux OS or "firstdisk" for ESXi.
+kvm                 Boolean          *optional*   The value is *true* or *false* to indicates to install kvm or not, default is **false**.
+switchDevices       Array            *optional*   **(ESXi only)** If specified, this contains an array of objects with switchName and uplinks (optional) parameters.  If *uplinks* is omitted, null or empty, the vswitch will be created with no uplinks.
```

Comment:
This is correct as is.  The vSwitch named 'switchName' will be created with no uplinks if the 'uplinks' parameter is omitted.  @anhou It may be worth additionally adding that if an uplink is attached to a vSwitch, it will be removed from the old vSwitch before being added to the vSwitch named by 'switchName'.